### DISCUSSION TITLE
SEO - h1 must be unique on all and across pages

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [CaiJimmy/hugo-theme-stack#869] [Q&A] [answered]
Hi again @CaiJimmy 

I'm back with the next question.

In your theme, we set in the `config/_default/config.toml` the title and description.

The issue is that it is used across all the pages in the `header` element inside the `aside` element. I see that in `layouts/partials/sidebar/left.html`.

It is fine for the homepage (e.g. `/`)

However, when you are on a post, page or list page (e.g. Category or Tag), the `h1` is used in two places on the same page :

- in the `header` element inside the `aside` element
- in the "layouts/_default/list.html"

**Issue 1**:  Best practices and accessibility standards recommend we should use one `h1` per page. Se [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements).

**Issue 2**: SEO best practices say we should us unique `h1` across pages on a same site. [See here](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls?hl=en&visit_id=638284882507299608-1675376168&rd=1), which I am sure you know.

So, how can we make sure:

- the homepage `h1` remains the one from `config/_default/config.toml`
- the list pages `h1` is the frontmatter `title` property

I tried to modified the template files, but I always fall into Issue 1 or 2...

Thanks for your additional help.

**Comment by CaiJimmy** [comment, marked as answer]
> I tried to modified the template files, but I always fall into Issue 1 or 2...

You can use conditionals to accomplish that. Something like:

```
{{ if .IsHome }}
    <h1>
{{ else }}
    <h2>
{{ end }}
```

**Reply by JeremieLitzler** [reply]
Thanks @CaiJimmy. I shall try that tomorrow. 

Where do I find the exhaustive list of the available variables/props like `.IsHome`?

I saw how to create local variables in the other discussion about shortcodes.

But I saw a lot of props like `. Inner` and so on.

Cheers!

**Reply by CaiJimmy** [reply]
> Where do I find the exhaustive list of the available variables/props like .IsHome?

Check Hugo documentation: https://gohugo.io/variables/page/

**Reply by JeremieLitzler** [reply]
Thanks @CaiJimmy.

The if ... else would have worked, but since:

1. `layouts/partials/article/components/header.html` and `layouts/partials/article/components/details.html` are shared with `layouts/partials/article/article.html` and `layouts/partials/article-list/default.html`
2. `IsHome` is not available in `layouts/partials/article/components/details.html`, 
 
I could only solve the issue with a duplicate template of  `layouts/partials/article/components/header.html` and `layouts/partials/article/components/details.html`.

**Comment by kininblog** [comment]
Thanks to Jeremie.
I duplicated details.html and header.html . This is quite easy to do, I wrote how to[ solve the h1 tag hugo stack theme](https://akazed.com/solusi-tag-h1-hugo-stack-theme/) on my blog
